tests/api/test_global_cli_context.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2024 Snowflake Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from dataclasses import fields
+
+from snowflake.cli.api.cli_global_context import (
+    _CONNECTION_CACHE,
+    _CliGlobalContextManager,
+    fork_cli_context,
+    get_cli_context,
+    get_cli_context_manager,
+)
+from snowflake.cli.api.output.formats import OutputFormat
+
+
+def test_reset_global_context_mgr():
+    """
+    Ensures that the reset() method is working properly.
+    """
+    mgr = _CliGlobalContextManager()
+    mgr.experimental = True
+    mgr.silent = True
+    mgr.output_format = OutputFormat.JSON
+    mgr.connection_context.database = "blahblah"
+    mgr.connection_context.password = "****"
+    mgr.override_project_definition = "project definition"
+    assert mgr.connection_cache == _CONNECTION_CACHE
+    mgr.connection_cache = None
+    mgr.reset()
+    assert mgr.connection_cache == _CONNECTION_CACHE
+    assert mgr.override_project_definition is None
+
+    initial_mgr = _CliGlobalContextManager()
+    for f in fields(mgr):
+        assert getattr(mgr, f.name) == getattr(
+            initial_mgr, f.name
+        ), f"{f.name} was not reset properly"
+
+
+# FIXME: needs asyncio test
```

Comment:
is this still relevant?

Reply:
Yes, we don't have the pytest-asyncio library yet. That will be introduced in the LSP PR, at which point I'll be able to write tests for the connection caching cleanup methods that are not currently exercised (because nothing creates an event loop, yet).